core/src/main/java/org/apache/iceberg/CatalogProperties.java

```diff
@@ -53,6 +56,57 @@ private CatalogProperties() {
   public static final long CACHE_EXPIRATION_INTERVAL_MS_DEFAULT = TimeUnit.SECONDS.toMillis(30);
   public static final long CACHE_EXPIRATION_INTERVAL_MS_OFF = -1;
 
+  /**
+   * When using {@link CachingFileIO} as io-impl, define the internal FileIO that is
+   * wrapped by CachingFileIO.
+   * <p>
+   * Default to {@link HadoopFileIO} if not defined.
+   */
+  public static final String CONTENT_CACHE_WRAPPED_FILE_IO_IMPL = "cache.content.wrapped-io-impl";
+  public static final String CONTENT_CACHE_WRAPPED_FILE_IO_IMPL_DEFAULT = HadoopFileIO.class.getName();
```

Comment:
I don't think that configuration needs to be this complicated. Instead of replacing the `FileIO` implementation and adding a second dynamic load, I would prefer using `cache-enabled=true` and handling this where we instantiate `FileIO`.

Reply:
`cache-enabled` is already taken by `CachingCatalog`

https://github.com/apache/iceberg/blob/ed54952b51a6cdc7d2350b8223c276421845a8e2/core/src/main/java/org/apache/iceberg/CatalogProperties.java#L39

Is it OK to enable both `CachingCatalog` and `CachingFileIO` by sharing this property key? Or should I make new key like `file-cache-enabled=true`?

Reply:
I think this would be an `io.cache-enabled` property.

Reply:
Done.